### PR TITLE
Complete all missing formatter implementations

### PR DESCRIPTION
**Pull request by mivano**
## Summary

Implements the two remaining stub methods that returned `Task.CompletedTask` without producing any output:

### TextOutputFormatter.WriteAccumulatedDiffCost
- Plain text diff output showing cost changes by service name, location, and resource group
- Source → target comparison with +/- change indicators
- Summary section with totals

### CsvOutputFormatter.WriteAccumulatedDiffCost  
- CSV export using CsvHelper with columns: Category, Name, SourceCost, TargetCost, Change, Currency
- Covers all three comparison dimensions (ServiceName, Location, ResourceGroup)
- Respects SkipHeader setting

### Tests
- 7 new tests in `FormatterGapTests.cs` covering:
  - Output is non-empty and contains expected content
  - Change signs (+/-) are present
  - Empty data edge case
  - Skip header behavior
  - All categories present in CSV output

### Notes
The other methods mentioned in the issue (MarkdownOutputFormatter.WriteCostByResource, WriteAnomalyDetectionResults, WriteDevTestComparison, and CsvOutputFormatter.WriteDevTestComparison) were already implemented on this branch.

### CODE REVIEW

File: src/OutputFormatters/TextOutputFormatter.cs [hunk 1]
```diff
@@ -238,8 +238,82 @@ public override Task WriteAnomalyDetectionResults(DetectAnomalySettings settings
...
+        var currency = "USD";
+        if (!settings.UseUSD)
+        {
+            var firstCost = accumulatedCostSource.Costs.FirstOrDefault();
+            if (firstCost != null && !string.IsNullOrEmpty(firstCost.Currency))
+            {
+                currency = firstCost.Currency;
+            }
+        }
+
```
[the remark this answers is not in the document]
Comment (mivano) [reply src/OutputFormatters/TextOutputFormatter.cs]: Fixed: Implemented a `GetPreferredCurrency()` local function that searches across all cost collections (Costs, ByServiceNameCosts, ByLocationCosts, ByResourceGroupCosts) from both source and target, returning the first non-empty currency found. Falls back to USD only if no currency is available anywhere.

File: src/OutputFormatters/TextOutputFormatter.cs [hunk 2]
```diff
@@ -238,8 +238,82 @@ public override Task WriteAnomalyDetectionResults(DetectAnomalySettings settings
     public override Task WriteAccumulatedDiffCost(DiffSettings settings, AccumulatedCostDetails accumulatedCostSource,
         AccumulatedCostDetails accumulatedCostTarget)
     {
+        var culture = CultureInfo.GetCultureInfo("en-US");
```
[the remark this answers is not in the document]
Comment (mivano) [reply src/OutputFormatters/TextOutputFormatter.cs]: Fixed: Changed from `CultureInfo.GetCultureInfo("en-US")` to `CultureInfo.CurrentCulture` for consistency with the rest of the formatter.

File: src/OutputFormatters/TextOutputFormatter.cs [hunk 3]
```diff
@@ -238,8 +238,82 @@ public override Task WriteAnomalyDetectionResults(DetectAnomalySettings settings
...
+            .ToList();
+
+        foreach (var item in allItems)
+        {
+            var sourceCost = sourceItems.Where(a => a.ItemName == item).Sum(a => useUSD ? a.CostUsd : a.Cost);
+            var targetCost = targetItems.Where(a => a.ItemName == item).Sum(a => useUSD ? a.CostUsd : a.Cost);
+            var diff = targetCost - sourceCost;
+            var diffSign = diff >= 0 ? "+" : "";
+            Console.WriteLine($"  {item}: {sourceCost.ToString("N2", culture)} -> {targetCost.ToString("N2", culture)} ({diffSign}{diff.ToString("N2", culture)}) {currency}");
+        }
```
[the remark this answers is not in the document]
Comment (mivano) [reply src/OutputFormatters/TextOutputFormatter.cs]: Fixed: Pre-aggregated costs into `sourceCosts` and `targetCosts` dictionaries using GroupBy/ToDictionary, then used dictionary lookups instead of repeated Where/Sum scans. This reduces the complexity from O(n²) to O(n).

File: src/OutputFormatters/CsvOutputFormatter.cs [hunk 4]
```diff
@@ -149,9 +149,40 @@ public override Task WritePricesPerRegion(WhatIfSettings settings,
...
+            .Union(targetItems.Select(a => a.ItemName))
+            .Distinct()
+            .ToList();
+
+        foreach (var name in allNames)
+        {
+            var sourceCost = sourceItems.Where(a => a.ItemName == name).Sum(a => useUSD ? a.CostUsd : a.Cost);
+            var targetCost = targetItems.Where(a => a.ItemName == name).Sum(a => useUSD ? a.CostUsd : a.Cost);
+            var diff = targetCost - sourceCost;
+            var currency = sourceItems.Concat(targetItems).FirstOrDefault(a => a.ItemName == name)?.Currency ?? "USD";
```
[the remark this answers is not in the document]
Comment (mivano) [reply src/OutputFormatters/CsvOutputFormatter.cs]: Fixed: Replaced the repeated Where/Sum pattern with GroupBy/ToDictionary pre-aggregation for costs and currencies, then iterate the union of keys. Also removes the redundant Distinct() (addressed in the other comment). Complexity reduced from O(n²) to O(n).

File: src/OutputFormatters/CsvOutputFormatter.cs [hunk 5]
```diff
@@ -149,9 +149,40 @@ public override Task WritePricesPerRegion(WhatIfSettings settings,
...
+
+        return ExportToCsv(settings.SkipHeader, diffRows);
     }
 
+    private static void AddDiffRows(List<object> rows, string category,
+        List<CostNamedItem> sourceItems, List<CostNamedItem> targetItems, bool useUSD)
+    {
+        var allNames = sourceItems.Select(a => a.ItemName)
+            .Union(targetItems.Select(a => a.ItemName))
+            .Distinct()
```
[the remark this answers is not in the document]
Comment (mivano) [reply src/OutputFormatters/CsvOutputFormatter.cs]: Fixed: Removed the redundant `.Distinct()` call since `.Union()` already returns distinct values. This was addressed as part of the broader AddDiffRows refactoring.

File: tests/AzureCostCli.Tests/OutputFormatters/FormatterGapTests.cs [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+    [Fact]
+    public async Task Text_WriteAccumulatedDiffCost_ProducesOutput()
+    {
+        var formatter = new TextOutputFormatter();
+        var source = CreateSampleDiffDetails(1.0);
+        var target = CreateSampleDiffDetails(1.2);
+        var settings = CreateDiffSettings();
+
+        var output = CaptureConsoleOutput(() => formatter.WriteAccumulatedDiffCost(settings, source, target));
+
```
[the remark this answers is not in the document]
Comment (mivano) [reply tests/AzureCostCli.Tests/OutputFormatters/FormatterGapTests.cs:66]: Fixed: Removed `async` from all 7 test methods, changing them from `async Task` to `void` since they don't use await. This eliminates the CS1998 warnings.
